server/src/infra/ldap_handler.rs

```diff
@@ -527,6 +544,47 @@ impl<Backend: BackendHandler + LoginHandler + OpaqueHandler> LdapHandler<Backend
         Ok(vec![make_add_error(LdapResultCode::Success, String::new())])
     }
 
+    async fn do_compare(&mut self, request: LdapCompareRequest) -> LdapResult<Vec<LdapOp>> {
+        let req = make_search_request::<String>(
+            &self.ldap_info.base_dn_str,
+            LdapFilter::And(vec![LdapFilter::Equality(
+                "dn".to_string(),
+                request.dn.to_string(),
+            )]),
+            vec![],
+        );
+        let res = self.do_search_or_dse(&req).await?;
+        let entry = res.first().ok_or(LdapError {
+            code: LdapResultCode::OperationsError,
+            message: "Search request returned nothing".to_string(),
+        })?;
```

Comment:
It might also be nice to check whether there's more than 1 result.

Reply:
I can do this. What shall be the error then?

Reply:
I also implemented this one with `OperationsError`. let me know what kind of error you prefer here.